### ISSUE TITLE
Optional quickfix format

### ISSUE THREAD

**Issue by FabioAntunes** [neomake/neomake#1834]
With the merging of #1693 the quickfix and the locations window will always have the height equal to the number os lines.

The idea would be using the same `vim-qf` option, making the resize optional.

> Automatically adjust the height of location/quickfix windows to 10 lines
> (Vim's default) or to the number of items in the list if that number is
> inferior to 10.
> 
> Add the line below to your vimrc to disable this feature: >
> 
>     let g:qf_auto_resize = 0

So this:
![screen shot 2018-01-25 at 11 09 54](https://user-images.githubusercontent.com/2544673/35391393-766b7dc0-01d6-11e8-9a11-43b685d85bbf.png)

Be again like this:
![screen shot 2018-01-25 at 13 48 53](https://user-images.githubusercontent.com/2544673/35391411-8ac71072-01d6-11e8-80af-3acf186a2e3b.png)

Unfortunately my vim script skills are very poor, but the fix would be probably as simple as changing this line

https://github.com/neomake/neomake/blob/928f33cf13dc2606a2a87c4babf9738e597ab13d/autoload/neomake.vim#L897

to something like
```git-diff
+ if  get(g:, 'some_variable', 1)
let height = min([len(a:loc_or_qflist), height])
+ endif
 ```

but this would probably break the tests and that's where my skills are not enough to fix them.

**Comment by blueyed** [comment]
See https://github.com/neomake/neomake/pull/1693#issuecomment-340856864.

We could have something like `let g:neomake_list_height = -X` to disable this feature, where X then would be used always.
A separate setting for applying the minimum might make sense though (but I would like to avoid adding settings).
Have to think about it..

Are you using `:set wrap` for the `qf` window(s)?  How do you set it?
Are you using vim-qf in parallel?
(there is also https://github.com/blueyed/vim-qf_resize).

**Comment by FabioAntunes** [comment]
Yes I'm using vim-qf in parallel, to autoclose the windows when I close the buffer related to it 

I set the  wrap by doing:
```
autocmd FileType qf setlocal wrap linebreak
```

What's the future plans? to add more features around qf windows to Neomake? It just seems that allowing other pluins to handle the qf and locations windows makes more sense and would create less conflicts in the future?

**Comment by blueyed** [comment]
> to add more features around qf windows to Neomake? 

Not really.
I agree that there are more specialized plugins that can do it, but it also makes sense to handle opening/closing them by default.
I am using a custom method myself to only open them in case of errors, and not warnings, which I plan to add to Neomake also.
Conflict-wise it should be good (we have a workaround for vim-qf though).

And given the custom-quickfix feature (`:call neomake#quickfix#enable()`) already, which needs a bit more polishing and docs, Neomake is quite good already - so nothing bigger planned.

Thinking about it however somet things come to mind:
1. make use of new qf-related features like the `context` and `size` properties.
2. queue for adding qf entries, to minimize BufRead and other autocmds.. this could make use of the `winid` property to see if it is visible, and the `efm`/`lines` feature to generate entries without modifying the current list, then either adding them altogether, or use `noautocmd` with all but the latest call in a make run.  Downside: no instant adding for faster makers, so not really good.
3. List of errors could be stored internally additionally (although that is done somehow already to display the error under the cursor).

> vim-qf in parallel, to autoclose the windows when I close the buffer related to it

That could be easily provided by Neomake probably, too.

Using vim-qf however, what about `let g:neomake_open_list = 0`?

**Comment by FabioAntunes** [comment]
If I use `let g:neomake_open_list = 0` then quickfix won't show when I have any linting issues

**Comment by blueyed** [comment]
Wouldn't `g:qf_auto_open_quickfix` / `g:qf_auto_open_loclist` do this for you then?
Have you changed those settings? (they are on by default IIRC)

**Comment by FabioAntunes** [comment]
Nop, I haven't changed the settings related to those two options. In fact this is my config for qf

```
let g:qf_window_bottom = 0
let g:qf_loclist_window_bottom = 0
let g:qf_auto_resize = 0
let g:qf_nowrap = 0
augroup qf
  autocmd!

  " automatically close corresponding loclist when quitting a window
  autocmd QuitPre * if &filetype != 'qf' | silent! lclose | endif
augroup END
```

Vim-qf just merged a pull request recently related to making the nowrap configurable. 
https://github.com/romainl/vim-qf/issues/63
